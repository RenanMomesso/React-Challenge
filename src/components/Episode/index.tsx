import { Link } from "react-router-dom";

export interface EpisodeListProps {
  serieNumber: number;
  serieName: string;
  serieDate: string;
  season: number;
}

const EpisodeList = ({
  serieNumber,
  serieName,
  serieDate,
  season,
}: EpisodeListProps) => {
  return (
    <tbody>
      <tr>
        <td>{season}</td>
        <td>
          <Link to={`/episode/${serieNumber}/${season}`}>{serieNumber}</Link>
        </td>
        <td>{serieName}</td>
        <td>{serieDate}</td>
      </tr>
    </tbody>
  );
};

export default EpisodeList;
