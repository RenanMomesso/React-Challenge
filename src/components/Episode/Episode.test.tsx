import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Episode from ".";

describe("Episode each list", () => {
  it("should render episode correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Episode
          serieDate="08/08/2016"
          serieName="Monster attack"
          serieNumber={5}
          season={1}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByRole("link", { name: "5" })).toBeInTheDocument();
    expect(screen.getByText("Monster attack")).toBeInTheDocument();
    expect(screen.getByText("08/08/2016")).toBeInTheDocument();
    expect(screen.getByText("1")).toBeInTheDocument();
  });
});
