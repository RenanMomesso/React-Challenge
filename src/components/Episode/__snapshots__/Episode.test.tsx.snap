// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode each list should render episode correctly 1`] = `
<div>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <a
          href="/episode/5/1"
        >
          5
        </a>
      </td>
      <td>
        Monster attack
      </td>
      <td>
        08/08/2016
      </td>
    </tr>
  </tbody>
</div>
`;
